src/ai_services.py

```
@@ -0,0 +1,315 @@
+        logger.info(f"[OPENAI] GROUP API call for group = {gid}, sender = {sender}")
+        resp = await openai.ChatCompletion.acreate(
+            model="gpt-4.1",
+            messages=messages,
+            functions=[SEARCH_FN, GET_HISTORY_FN, SEND_PVT_MSG_FN],
+            function_call="auto",
+            temperature=0.7,
+            max_tokens=300
+        )
+        
+    raw = resp.choices[0].message.content or ""
+    raw = raw.strip().strip("```").strip()
```

Ruff (0.11.9)

Using `.strip()` with multi-character strings is misleading

(B005)